pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +333,83 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{
+		LabelSelector: label.TopologyNetwork.Name,
+	})
+	registerHandlers[controllers.Object](c, n.gatewayResourceClient, "Gateways", n.handleGatewayResource, nil)
```

Comment:
```suggestion
	registerHandlers[*v1beta1.Gateway(c, n.gatewayResourceClient, "Gateways", n.handleGatewayResource, nil)
```

then you don't need to cast

Reply:
AFAICT, I can't convert `kclient.Untyped` to an `Informer[*v1beta1.Gateway]` (or anything that changes the generic)